network/components/virt_rx.c

```diff
@@ -99,7 +99,9 @@ void rx_return(void)
             // usermode CleanInvalidate (faster than a Invalidate via syscall).
             //
             // [1]: https://developer.arm.com/documentation/ddi0595/2021-06/AArch64-Instructions/DC-IVAC--Data-or-unified-Cache-line-Invalidate-by-VA-to-PoC
-            cache_clean_and_invalidate(buffer_vaddr, buffer_vaddr + buffer.len);
+            #ifdef CONFIG_ARCH_ARM
+            cache_clean_and_invalidate(buffer_vaddr, buffer_vaddr + ROUND_UP(buffer.len, 1 << CONFIG_L1_CACHE_LINE_SIZE_BITS));
+            #endif
```

Comment:
Why is this not needed?

Reply:
Because RISC-V, in general but not all cases I believe, is DMA coherent.